src/x509.c

```diff
@@ -5195,840 +5195,1077 @@ int wolfSSL_X509_cmp(const WOLFSSL_X509 *a, const WOLFSSL_X509 *b)
+        && x509->altNames != NULL) {
+        entry = x509->altNames;
 
-        if (bio == NULL || x509 == NULL) {
-            return WOLFSSL_FAILURE;
+        wolfSSL_BIO_write(bio, "            X509v3 Subject Alternative Name: ",
+            (int)XSTRLEN("            X509v3 Subject Alternative Name: "));
+        if (x509->subjAltNameCrit) {
+            wolfSSL_BIO_write(bio, "critical", (int)XSTRLEN("critical"));
         }
+        wolfSSL_BIO_write(bio, "\n", 1);
+        wolfSSL_BIO_write(bio, "                ",
+            (int)XSTRLEN("                "));
 
-        if (wolfSSL_BIO_write(bio, "Certificate:\n",
-                      (int)XSTRLEN("Certificate:\n")) <= 0) {
-                return WOLFSSL_FAILURE;
+        while (entry != NULL) {
+            ++nameCount;
+            if (nameCount > 1) {
+                wolfSSL_BIO_write(bio, ", ", 2);
+            }
+
+            if (entry->type == ASN_DNS_TYPE) {
+                if (entry->name == NULL) {
+                    WOLFSSL_MSG("NULL DNS alt name.");
+                    return WOLFSSL_FAILURE;
+                }
+                wolfSSL_BIO_write(bio, "DNS:", 4);
+                wolfSSL_BIO_write(bio, entry->name, XSTRLEN(entry->name));
+            }
+        #if defined(OPENSSL_ALL) || defined(WOLFSSL_IP_ALT_NAME)
+            else if (entry->type == ASN_IP_TYPE) {
+                if (entry->ipString == NULL) {
+                    WOLFSSL_MSG("NULL IP address alt name.");
+                    return WOLFSSL_FAILURE;
+                }
+                wolfSSL_BIO_write(bio, "IP Address:", 11);
+                wolfSSL_BIO_write(bio, entry->ipString,
+                                  XSTRLEN(entry->ipString));
+            }
+        #endif /* OPENSSL_ALL || WOLFSSL_IP_ALT_NAME */
+            else {
+                WOLFSSL_MSG("Bad alt name type.");
+                ret = WOLFSSL_FAILURE;
+                break;
+            }
+
+            entry = entry->next;
         }
 
-        if (wolfSSL_BIO_write(bio, "    Data:\n",
-                      (int)XSTRLEN("    Data:\n")) <= 0) {
-                return WOLFSSL_FAILURE;
+        if (ret == WOLFSSL_SUCCESS) {
+            wolfSSL_BIO_write(bio, "\n", 1);
         }
+    }
 
-        /* print version of cert */
-        {
-            int version;
-            char tmp[20];
+    return ret;
+}
 
-            if ((version = wolfSSL_X509_version(x509)) < 0) {
-                WOLFSSL_MSG("Error getting X509 version");
-                return WOLFSSL_FAILURE;
-            }
-            if (wolfSSL_BIO_write(bio, "        Version:",
-                          (int)XSTRLEN("        Version:")) <= 0) {
-                return WOLFSSL_FAILURE;
+#ifdef XSNPRINTF
+static int X509PrintKeyUsage(WOLFSSL_BIO* bio, WOLFSSL_X509* x509)
+{
+    int ret = WOLFSSL_SUCCESS;
+    /*
+     * 162 = Leading indent (16) + all usageStrs concatenated with
+     * commas + \n + \0.
+     */
+    char tmp[162];
```

Comment:
Technically violates the small stack rule (100 bytes), however anyone using this level of compat function should not be needing/using small stack. Same for a few others below.

Reply:
Looks like we were typing comments around the same time, but my head was in the same space, yeah. See my comment about rewriting to not use these temporary buffers.